### PR TITLE
[QuarkusClassLoaderHandler] Check for both `elements` and split element fields to accomdate Quarkus 3.15.x and earlier and 3.16.x and later.

### LINKED ISSUE #891: Webjars broken with Quarkus 3.16 due to an exception in Classgraph
In code that worked with Quarkus 3.15, I am now getting the following exception after upgrading to Quarkus 3.16:

    java.lang.RuntimeException: Exception while calling findClassLoaderOrder for nonapi.io.github.classgraph.classloaderhandler.QuarkusClassLoaderHandler
	at nonapi.io.github.classgraph.classloaderhandler.ClassLoaderHandlerRegistry$ClassLoaderHandlerRegistryEntry.findClasspathOrder(ClassLoaderHandlerRegistry.java:239)
	at nonapi.io.github.classgraph.classpath.ClasspathFinder.<init>(ClasspathFinder.java:277)
	at io.github.classgraph.Scanner.<init>(Scanner.java:170)
	at io.github.classgraph.ClassGraph.scanAsync(ClassGraph.java:1562)
	at io.github.classgraph.ClassGraph.scanAsync(ClassGraph.java:1590)
	at io.github.classgraph.ClassGraph.scan(ClassGraph.java:1615)
	at io.github.classgraph.ClassGraph.scan(ClassGraph.java:1654)
	at io.github.classgraph.ClassGraph.scan(ClassGraph.java:1667)
	at org.webjars.WebJarAssetLocator.scanForWebJars(WebJarAssetLocator.java:188)
	at org.webjars.WebJarAssetLocator.<init>(WebJarAssetLocator.java:210)
	at org.webjars.WebJarAssetLocator.<init>(WebJarAssetLocator.java:198)
	...
    Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:115)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at nonapi.io.github.classgraph.classloaderhandler.ClassLoaderHandlerRegistry$ClassLoaderHandlerRegistryEntry.findClasspathOrder(ClassLoaderHandlerRegistry.java:236)
	... 21 more
    Caused by: java.lang.NullPointerException: Cannot invoke "java.util.Collection.iterator()" because the return value of "nonapi.io.github.classgraph.reflection.ReflectionUtils.getFieldVal(boolean, Object, String)" is null
	at nonapi.io.github.classgraph.classloaderhandler.QuarkusClassLoaderHandler.findClasspathOrderForQuarkusClassloader(QuarkusClassLoaderHandler.java:120)
	at nonapi.io.github.classgraph.classloaderhandler.QuarkusClassLoaderHandler.findClasspathOrder(QuarkusClassLoaderHandler.java:111)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	... 23 more

The following code in Classgraph seems to trigger the exception:

    for (final Object element : (Collection<Object>) classpathOrder.reflectionUtils.getFieldVal(false,
                classLoader, "elements")) {
         ...
    }

I'm assuming some behavior of the Quarkus ClassLoader changed in 3.16? I do not know any details about this though.

I attempted manually forcing the latest version of Classgraph (4.8.177), but exception persists.

    $ java -version
    openjdk version "21.0.5" 2024-10-15 LTS
    OpenJDK Runtime Environment Temurin-21.0.5+11 (build 21.0.5+11-LTS)
    OpenJDK 64-Bit Server VM Temurin-21.0.5+11 (build 21.0.5+11-LTS, mixed mode, sharing)

### PR DESCRIPTION
**Pull request by michael-simons**
Fixes #891 (Sorry, typo and tagged the wrong issue)

### CODE REVIEW

File: src/main/java/nonapi/io/github/classgraph/classloaderhandler/QuarkusClassLoaderHandler.java [hunk 1]
```diff
@@ -117,8 +118,17 @@ public static void findClasspathOrder(final ClassLoader classLoader, final Class
...
             final ClasspathOrder classpathOrder, final ScanSpec scanSpec, final LogNode log) {
-        for (final Object element : (Collection<Object>) classpathOrder.reflectionUtils.getFieldVal(false,
-                classLoader, "elements")) {
+
+        Collection<Object> elements = (Collection<Object>) classpathOrder.reflectionUtils.getFieldVal(false,
+            classLoader, "elements");
+        if(elements == null) {
+            elements = new ArrayList<>();
+            for(String fieldName : new String[] {"normalPriorityElements", "lesserPriorityElements"}) {
+                elements.addAll((Collection<Object>) classpathOrder.reflectionUtils.getFieldVal(false,
```
Comment (lukehutch) [inline src/main/java/nonapi/io/github/classgraph/classloaderhandler/QuarkusClassLoaderHandler.java]: Please check if `getFieldVal` returns null before calling `addAll` here.

**Comment by lukehutch** [comment]
Other than the one change I requested, looks good -- thank you, @michael-simons!

File: src/main/java/nonapi/io/github/classgraph/classloaderhandler/QuarkusClassLoaderHandler.java (hunk 1 above)
Comment (michael-simons) [reply src/main/java/nonapi/io/github/classgraph/classloaderhandler/QuarkusClassLoaderHandler.java]: Yeah, good idea :)

**Comment by michael-simons** [comment]
Thanks, @lukehutch, change is done.

**Comment by lukehutch** [comment]
Excellent, thank you @michael-simons! Merging.

You said in the issue that you would write the full URL handler to handle in-memory classes. I assume that will build on this PR. I will wait until that is available to push out a release. Thanks for volunteering to do that work too.

**Comment by michael-simons** [comment]
To help OGM users I’d love to see this go out first (also less pressure on me to get this right) if that’s ok with you.

**Comment by lukehutch** [comment]
@michael-simons Released in 4.8.178 (and I'm releasing another unrelated PR that I forgot about in 4.8.179). 

PS Narcissus still has build issues (I opened a new issue there), but I decided to just push out ClassGraph without those fixes, because you need it quickly!

**Pull request merged**